rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -100,11 +116,15 @@ def __str__(self) -> str:
         )
 
 
-class BaseAcquisitionsOrder(metaclass=ABCMeta):
-    _acquisitions: deque[AssetAcquisitionEvent]
+class BaseCostBasisMethod(metaclass=ABCMeta):
+    # For FIFO, the time at the point of insertion is used
+    # For LIFO, the time at the point of insertion is used with a negation
+    # For HIFO, the amount of the acquisition event is used with a negation
+    # The reason for the negation is because of Python's `heapq` implementation
+    _acquisitions: list[Tuple[ACQUISITION_PRIORITY, AssetAcquisitionEvent]]
 
     def __init__(self) -> None:
-        self._acquisitions = deque()
+        self._acquisitions = []
```

Comment:
for the same reason as yabir mentioned you should absolutely change the name here. 
```suggestion
        self._acquisitions_heap = []
```